integr8ly/ci-cd
# changed pipeline to generate reports despite of test failures

## What & Why & Motivation
browser-based-testsuite-pipeline job build result should be UNSTABLE in case of test errors/failures. Not failure.

## How
shell script that runs the tests always exits with 0. Output of the script is parsed to see whether there were test failures or errors. Based on that the build result is set. In case of error the publishing is skipped (no xml report is generated in that case anyway)

## Verification Steps
Go to this pipeline (with the definition the same as the target pipeline): https://integreatly-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/ppaszki-browser-based/7/console and check the output to confirm that the results are published

## Checklist:


## Progress
